web/services/hosts_next.go

```diff
@@ -87,3 +105,28 @@ func (s *hostsNextService) GetAllTags() ([]string, error) {
 
 	return tags, nil
 }
+
+func (s *hostsNextService) Heartbeat(agentID string) error {
+	heartbeat := &entities.HostHeartbeat{
+		AgentID: agentID,
+	}
+
+	return s.db.Clauses(clause.OnConflict{
+		Columns: []clause.Column{
+			{Name: "agent_id"},
+		},
+		DoUpdates: clause.AssignmentColumns([]string{"updated_at"}),
+	}).Create(heartbeat).Error
+}
+
+func computeHealth(host *entities.Host) string {
```

Comment:
`computeHealth` sounds a li'l clunky, maybe `getHealth`?

Reply:
it is a computation based on some other values (in the future will probably aggregate stuff from other sources). i will think of a better name.